pkg/applicationserver/applicationserver.go

```diff
@@ -1016,7 +1023,7 @@ func (as *ApplicationServer) handleJoinAccept(ctx context.Context, ids *ttnpb.En
 				SkipSessionKeyIDs: [][]byte{items[0].SessionKeyId},
 				ResultFunc: func(decrypted, _ []*ttnpb.ApplicationDownlink, err error) {
 					if err != nil {
-						as.registerDropDownlinks(ctx, ids, decrypted, err)
+						as.registerDropDownlinks(ctx, dev.Ids, decrypted, receivedAt, err)
```

Comment:
This `receivedAt` is slightly misleading - it is the time at which the Join Accept which contains this invalidate downlink was received by the Application Server, not the original timestamp at which the downlink was enqueued.

Do we want to keep using the `nil` timestamps, because we cannot recover the original timestamp, or is it fine to always fill it with the closest timestamp to the message ?

Reply:
I think it's fine to set it to the timestamp of a message that triggered the event, for application layer ordering purposes.